Automattic/vip-go-mu-plugins
# Fix VIP stats functions when Jetpack stats module is disabled

## Description

Supersedes https://github.com/Automattic/vip-go-mu-plugins/pull/2266.

The functions `wpcom_vip_top_posts_array` and `wpcom_vip_get_post_pageviews` would fail if the Jetpack stats module was disabled. We now check if it is enabled before proceeding.

In the cases in which Jetpack was not found, we would return _fake_ data. We now return empty values and we log the error.

## Changelog Description

### Fix VIP stats functions when Jetpack stats module is disabled

This functions now return data.

## Checklist

Please make sure the items below have been covered before requesting a review:


## Steps to Test

_Taken from Raam's PR_

1. Before applying this patch,  programmatically disable the Jetpack Stats module and then use `wpcom_vip_top_posts_array()` or `wpcom_vip_get_post_pageviews()`. The following code snippet can be added to an MU plugin for testing:

```php
<?php
// Disable the Stats Jetpack Module
function disable_jetpack_modules( $modules ) {
	$disabled_modules = ['stats'];

	foreach ( $disabled_modules as $module_slug ) {
		$found = array_search( $module_slug, $modules, true );
		if ( false !== $found ) {
			unset( $modules[ $found ] );
		}
	}
	return $modules;
}
add_filter( 'jetpack_active_modules', 'disable_jetpack_modules', 99, 9 );

// Use wpcom_vip_get_post_pageviews() to show Post Views at the bottom of Single Posts content
function filter_the_content_in_the_main_loop( $content ) {
    if ( is_singular() && in_the_loop() && is_main_query() ) {
        return $content . esc_html__( 'Post Views: ' . wpcom_vip_get_post_pageviews( get_the_ID(), 1 ) );
    }

    return $content;
}
add_filter( 'the_content', 'filter_the_content_in_the_main_loop', 1 );
```
2. Verify that this PHP Fatal error is thrown when viewing a single post:
```
PHP Fatal error:  Uncaught Error: Call to undefined function stats_get_csv()
```
3. Apply the patch in this PR and verify that the error is no longer thrown (the functions silently fall back to using fake data as expected).
